engine/access/ingestion/engine_test.go

```diff
@@ -144,9 +175,12 @@ func (s *Suite) SetupTest() {
 
 // TestOnFinalizedBlock checks that when a block is received, a request for each individual collection is made
 func (s *Suite) TestOnFinalizedBlock() {
```

Comment:
can you add a test that covers the case where the engine had run before (the progress consumer was initialized) and it catches up a few blocks on startup?

Reply:
one option for testing that each block is attempted is to check which blocks are passed to `IndexBlockForCollections` using a blocks mock. The test could go something like:
1. Do the normal test
2. Create a new instance of engine with a new instance of the consumer progress using the same db
3. Call `OnFinalizedBlock` with a block several blocks newer than the last block processed
4. Make sure that `IndexBlockForCollections` is called with the next block

Reply:
[86e7b38](https://github.com/onflow/flow-go/pull/5714/commits/86e7b382a3812d0229e864bf4332c2123d424df2)